src/handlers/expiry/providers/TestDataProvider.ts

```diff
@@ -22,61 +17,66 @@ export class TestDataProvider implements ITestDataProvider {
     try {
       this.testResultsDAO = this.testResultsDAO as TestResultsDAO;
       const data = await this.testResultsDAO.getBySystemNumber(systemNumber);
+      console.log(`getTestHistory: Data Count -> ${data?.Count}`);
       if (data?.Count) {
         result = data.Items as ITestResult[];
         return result
-          .filter((test) => test.testStatus === TEST_STATUS.SUBMITTED)
+          .filter((test) => test.testStatus === enums.TEST_STATUS.SUBMITTED)
           .filter(
             (testResult: {
               testStartTimestamp: string | number | Date;
               testEndTimestamp: string | number | Date;
             }) => {
-              return (
-                moment(testResult.testStartTimestamp).isAfter(fromDateTime) &&
-                moment(testResult.testEndTimestamp).isBefore(toDateTime)
+              const { testStartTimestamp, testEndTimestamp } = testResult;
+              if (!DateProvider.isValidDate(testStartTimestamp) || !DateProvider.isValidDate(testEndTimestamp)) {
+                throw new Error(`Invalid timestamp -> StartTimestamp: ${testStartTimestamp} EndTimeStamp: ${testEndTimestamp} `);
+              }
+              return DateProvider.isBetweenDates(
+                testStartTimestamp,
+                testEndTimestamp,
+                fromDateTime,
+                toDateTime
               );
             }
           );
       }
       return result;
     } catch (error) {
-      if (!(error instanceof HTTPError)) {
-        console.log(error);
-        error = new HTTPError(500, MESSAGES.INTERNAL_SERVER_ERROR);
-      }
+      console.error("getTestHistory: Error ", error);
+      error = new HTTPError(500, enums.MESSAGES.INTERNAL_SERVER_ERROR);
```

Comment:
getTestHistory is called by getMostRecentExpiryDate which is called by the controller (generateExpiryDate()) which is already catching the error and throwing 500 if it fails (or the DAO fails too) with async, I don't think it will be necessary to check on HTTPError however do we want to be more meaningful about the error we return to differentiate between a service failure or failure to calculate the date because of the data we have received?

Reply:
revert the change.